plugins/json/core/src/main/java/org/pentaho/di/trans/steps/jsoninput/JsonInput.java

```diff
@@ -326,6 +327,12 @@ public JSONObject selectFieldsAction( Map<String, String> queryParamToValues ) {
     try {
       JsonInputMeta jsonInputMeta = (JsonInputMeta) getStepMetaInterface();
       FileInputList fileInputList = jsonInputMeta.getFiles( getTransMeta().getBowl(), getTransMeta() );
+        if ( fileInputList == null || fileInputList.getFiles().isEmpty() ) {
```

Comment:
If `fileInputList.getFiles()` is null NPE will be thrown
Hence instead of `fileInputList.getFiles().isEmpty()` you might want to use `CollectionUtils.isEmpty( fileInputList.getFiles() )`